### ISSUE TITLE
type() segfaults when incorrect argument is passed

### ISSUE THREAD

**Issue by peterhinch** [micropython/micropython#3812]
On Unix and Pyboard:
```python
class Foo:
    pass
type('Bar', Foo, {})  # Should be (Foo,)
```

**Comment by dpgeorge** [comment]
Thanks for the report.  Looks like this is fixed by the (unmerged) PR #3691.

**Comment by dpgeorge** [comment]
Should be fixed by 05b13fd292b42f20affc7cae218d92447efbf6d6

**Comment by peterhinch** [comment]
Indeed, now fixed:
```
MicroPython v1.9.3-736-gc60589c on 2018-05-30; linux version
Use Ctrl-D to exit, Ctrl-E for paste mode
>>> 
paste mode; Ctrl-C to cancel, Ctrl-D to finish
=== class Foo:
===     pass
=== type('Bar', Foo, {})  # Should be (Foo,)
=== 
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
TypeError: 
>>> 
```
